hazelcast/src/test/java/com/hazelcast/client/impl/proxy/DistributedObjectListenerTest.java

```diff
@@ -116,6 +117,22 @@ public void testGetDistributedObjectsAfterRemove_fromClient() {
         assertEquals(1, client2.getDistributedObjects().size());
     }
 
+    @Test
+    public void getDistributedObjects_ShouldNotRecreateProxy_AfterDestroy() {
+        HazelcastInstance member = hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        Future destroyProxyFuture = spawn(() -> {
+            for (int i = 0; i < 1000; i++) {
+                IMap<Object, Object> map = member.getMap("map-" + i);
+                map.destroy();
+            }
+        });
+        while (!destroyProxyFuture.isDone()) {
+            client.getDistributedObjects();
```

Comment:
do you really want a busy loop here or a small sleep between tries?

Reply:
If I add a sleep of 10 milliseconds, I can catch 0 to 2 recreated proxies with the old behaviour. (Without any sleep its around 100) I think this way, the test is much more reliable